bug: `using std` causes misleading error for undefined types like i512

## Summary

When any `using` directive is present, undefined types that look like sized integers (e.g., `i512`) produce a misleading "type mismatch" error instead of "undefined type".

## Steps to Reproduce

**Without `using`** (correct behavior):
```ez
do main() {
    temp x i512 = 123
}
```

```
error[E3008]: undefined type 'i512'
  --> file.ez:2:10
   |
2  |     temp x i512 = 123
   |          ^ type is not defined
```

**With any `using` directive** (incorrect behavior):
```ez
import @std
using std

do main() {
    temp x i512 = 123
}
```

```
error[E3001]: type mismatch: cannot assign int to i512
  --> file.ez:5:10
   |
5  |     temp x i512 = 123
   |          ^ types do not match
```

## Additional Testing

- Happens with `using std`, `using crypto`, `using math` - any `using` directive
- Does NOT happen with just `import @module` (no `using`)
- Other invalid types like `foobar`, `i7`, `i100`, `i1024` correctly show "undefined type" even with `using`
- Issue appears specific to `i512` pattern

## Expected Behavior

Both cases should show `error[E3008]: undefined type 'i512'`.

## Root Cause

The `using` directive appears to affect type resolution in a way that causes `i512` to be treated as a valid type instead of undefined. This may be related to #909 (using directive scope issues).

## Environment

- EZ version: dev (tested on main branch)

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -396,6 +396,13 @@ func (tc *TypeChecker) TypeExists(typeName string) bool {
 		return true
 	}
 
+	// Check if the type looks like a built-in sized type pattern (i<N>, u<N>, f<N>)
+	// If it matches the pattern but wasn't found in tc.types, it's definitely invalid
+	// and should NOT be deferred to module resolution
+	if looksLikeBuiltinSizedType(typeName) {
+		return false
+	}
+
 	// Check if the type might be available via file-level 'using' directive
 	// For unqualified type names, if a module is imported via 'using',
 	// the type will be validated at runtime when the module is loaded
@@ -419,6 +426,31 @@ func (tc *TypeChecker) TypeExists(typeName string) bool {
 	}
 
 	return false
+}
+
+// looksLikeBuiltinSizedType checks if a type name follows the pattern of built-in
+// sized types (i8, i16, u32, f64, etc.) but is NOT a valid one.
+// This prevents invalid types like i512 or u1024 from being deferred to module resolution.
+func looksLikeBuiltinSizedType(typeName string) bool {
+	if len(typeName) < 2 {
+		return false
+	}
+
+	prefix := typeName[0]
+	if prefix != 'i' && prefix != 'u' && prefix != 'f' {
+		return false
+	}
+
+	// Check if the rest is all digits
+	for _, c := range typeName[1:] {
+		if c < '0' || c > '9' {
+			return false
+		}
+	}
+
+	// It matches the pattern (e.g., i512, u1024, f128)
+	// If we got here, it wasn't found in tc.types, so it's an invalid sized type
+	return true
 }
 
 // RegisterType adds a user-defined type to the registry

